library/module_utils/oneview.py

```diff
@@ -344,8 +344,233 @@ class OneViewModuleBase(object):
+
+    def _build_argument_spec(self, additional_arg_spec, validate_etag_support):
+
+        merged_arg_spec = dict()
+        merged_arg_spec.update(self.ONEVIEW_COMMON_ARGS)
+
+        if validate_etag_support:
+            merged_arg_spec.update(self.ONEVIEW_VALIDATE_ETAG_ARGS)
+
+        if additional_arg_spec:
+            merged_arg_spec.update(additional_arg_spec)
+
+        return merged_arg_spec
+
+    def _check_hpe_oneview_sdk(self):
+        if not HAS_HPE_ONEVIEW:
+            self.module.fail_json(msg=self.HPE_ONEVIEW_SDK_REQUIRED)
+
+    def _create_oneview_client(self):
+        if self.module.params.get('hostname'):
+            config = dict(ip=self.module.params['hostname'],
+                          credentials=dict(userName=self.module.params['username'], password=self.module.params['password'],
+                                           authLoginDomain=self.module.params.get('auth_login_domain', '')),
+                          api_version=self.module.params['api_version'],
+                          image_streamer_ip=self.module.params['image_streamer_hostname'])
+            self.oneview_client = OneViewClient(config)
+        elif not self.module.params['config']:
+            self.oneview_client = OneViewClient.from_environment_variables()
+        else:
+            self.oneview_client = OneViewClient.from_json_file(self.module.params['config'])
+
+    def set_resource_object(self, resource_client):
+        self.resource_client = resource_client
+        name = None
+
+        if self.data and self.data.get("name"):
+            name = self.data["name"]
+
+        if not name and self.module.params.get("name"):
+            name = self.module.params["name"]
+
+        if name:
+            self.current_resource = self.resource_client.get_by_name(name)
+
+    @abc.abstractmethod
+    def execute_module(self):
+        """
+        Abstract method, must be implemented by the inheritor.
+
+        This method is called from the run method. It should contain the module logic
+
+        :return: dict: It must return a dictionary with the attributes for the module result,
+            such as ansible_facts, msg and changed.
+        """
+        pass
+
+    def run(self):
+        """
+        Common implementation of the OneView run modules.
+
+        It calls the inheritor 'execute_module' function and sends the return to the Ansible.
+
+        It handles any OneViewModuleException in order to signal a failure to Ansible, with a descriptive error message.
+
+        """
+        try:
+            if self.validate_etag_support:
+                if not self.module.params.get('validate_etag'):
+                    self.oneview_client.connection.disable_etag_validation()
+
+            result = self.execute_module()
+
+            if not result:
+                result = {}
+
+            if "changed" not in result:
+                result['changed'] = False
+
+            self.module.exit_json(**result)
```

Comment:
Better it can add in the 'else' case of  'try/except/else'  .  If there is no exception then  self.module.exit_json(**result)

Reply:
We cannot keep self.module.exit_json in else part as self.module.exit itself can throw an exception if the result is not a valid JSON.